pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -620,6 +614,8 @@ func applyTrafficPolicy(opts buildClusterOpts) {
 		applyH2Upgrade(opts, connectionPool)
 		applyOutlierDetection(opts.cluster, outlierDetection)
 		applyLoadBalancer(opts.cluster, loadBalancer, opts.port, opts.proxy, opts.mesh)
+	} else if opts.cluster.GetType() == cluster.Cluster_ORIGINAL_DST {
```

Comment:
Should we move this to buildDefaultCluster and set the LB when cluster type is ORIGINAL_DST?

Reply:
It feels slightly incosnsitent to apply LbPolicy here for outbound but in buildDefaultCluster in for inbound, which is why I put it here. I don't mind either way though, the end result should be the same?

Reply:
It should be same. I am suggesting we move it for both inbound and outbound to buildDefaultCluster for ORIGINAL_DST and remove this special handling https://github.com/istio/istio/blob/e15b5c7e12cfe5de337b82a7c91b1de5c8b2cee4/pilot/pkg/networking/core/v1alpha3/cluster.go#L803 at both places as it can not change.